File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    public int solution(int[] numbers, int target) {
        int num_ln = numbers.length;
        int result_ln = (int)Math.pow(2,num_ln);
        int[] result = new int[result_ln];
        int answer = 0;
        for(int i = 0; i < result_ln; i++){
            for(int j = 0; j < num_ln; j++){
                result[i] += (int) Math.pow(-1,i/(result_ln/(int)Math.pow(2,j+1)))*numbers[j];
            }
            if(result[i] == target) answer++;
        }
        return answer;
    }
}

File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    var answer = 0;
    array = [];
    const n = 2**numbers.length;
    for(let i = 0; i < n; i++){
        array[i] = 0;
        for(let j = 0; j < numbers.length; j++){
            array[i] += (Math.floor(i/(2**(numbers.length -j - 1)))%2 === 0?1:-1)*numbers[j];
        }
    }
    array.forEach((a)=>{
        if(a === target)
            answer++;
    });
    return answer;
}